src/Module.ts

```diff
@@ -735,32 +735,36 @@ export default class Module {
 		timeStart('analyse ast', 3);
 
 		this.astContext = {
-			addDynamicImport: this.addDynamicImport.bind(this),
-			addExport: this.addExport.bind(this),
-			addImport: this.addImport.bind(this),
-			addImportMeta: this.addImportMeta.bind(this),
+			addDynamicImport: (node: ImportExpression): void => this.addDynamicImport(node),
+			addExport: (
+				node: ExportAllDeclaration | ExportNamedDeclaration | ExportDefaultDeclaration
+			): void => this.addExport(node),
+			addImport: (node: ImportDeclaration): void => this.addImport(node),
+			addImportMeta: (node: MetaProperty): void => this.addImportMeta(node),
```

Comment:
What is the advantage of doing this? Considering performance, by understanding is that `.bind` is now more efficient as it avoids the indirection of the additional function, and these functions are "hot" in that they are called very often during the tree-shaking phase.

Reply:
yeah, that has nothing to do with typescript. there were perf issues with bind (in the past), I think those have been fixed tho. 

it's just consistency, e.g.: https://github.com/rollup/rollup/pull/4361/files/437e90bc8e8e784d8b3667280d4890dbcfff9945#diff-bdf37a1325be24e5865c89e911c2f1125a1d857ec429afb790bde572bbe12b0fR761-R765

but I'll revert.